src/System.Linq/src/System/Linq/Where.cs

```diff
@@ -320,6 +458,8 @@ public override Iterator<TResult> Clone()
                 return new WhereSelectListIterator<TSource, TResult>(_source, _predicate, _selector);
             }
 
+            public int GetCount(bool onlyIfCheap) => onlyIfCheap ? -1 : EnumerableHelpers.Count(this);
```

Comment:
@stephentoub, I wasn't sure that `GetCount` was worth optimizing; `e.Count(pred)` is typically preferred over `e.Where(pred).Count()`, and optimizing this function would involve some imperative code (maintaining an `int` variable we would increment every time a predicate was hit).

Thinking about it more though, someone may pass `e.Where(...)` to a function that invokes `Count()` on the enumerable, so I guess writing this inline may be worthwhile.